### PR TITLE
First pass at caching identities

### PR DESCRIPTION
**Pull request by aphelionz**
This is a very, very simple approach to caching identities created by the `static async Identities.createIdentity` function. It uses a deterministically stringified options object as the cache key and the `identity` object as the value.

This ensures that identical objects will return an identity instead of creating a new one, provided that it exists in the cache.

Feedback welcome, as always!

### CODE REVIEW

File: src/identities.js [hunk 1]
```diff
@@ -74,6 +77,9 @@ class Identities {
   }
 
   static async createIdentity (options = {}) {
+    const cacheKey = stringify(options)
```
Comment (haadcode) [inline src/identities.js]: What happens if a `Keystore` instance is passed as `options.keystore`?

Comment (haadcode) [reply src/identities.js]: Also, would it make more sense to use the actual `id` as the cache key?

Comment (aphelionz) [reply src/identities.js]: That was the first thing I tried, but `id` is not passed consistently in practice and waiting until after the identity is created defeats the purpose.

Deterministic stringify works but agree it's messy. Hashing the object could work?

**Comment by aphelionz** [comment]
Good experiment but a better solution was suggested

**Pull request closed without merging**